src/SdCard.cpp

```diff
@@ -146,14 +155,15 @@ char *SdCard_pickRandomSubdirectory(char *_directory) {
 			break;
 		}
 		if (!fileItem.isDirectory()) {
+			fileItem.close();
 			continue;
 		} else {
 			#if ESP_ARDUINO_VERSION_MAJOR >= 2
 				strncpy(buffer, (char *) fileItem.path(), 255);
 			#else
 				strncpy(buffer, (char *) fileItem.name(), 255);
 			#endif
-
+			fileItem.close();
```

Comment:
It's not necessary to call `fileItem.close()` here. `fileItem` is a local variable, so it's destructor is called when it get's out of scope (this is at the end of the loop).
The File Object has a `std::shared_ptr<FileImpl> p` holding a pointer to the file system object. As soon as the File object is destroyed, the smart pointer has 0 references to it and it in turn calls the destructor of the object it holds. This calls the virtual destructor `FileImpl::~FileImpl()` and this is forwarded to the implementation `VFSFileImpl ::~VFSFileImpl()` which calls close on the file object [here](https://github.com/espressif/arduino-esp32/blob/7693f7410a1885f8345be377b8197f60b5cd93f6/libraries/FS/src/vfs_api.cpp#L345).